components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -429,6 +432,28 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         }
 
         await this.userDB.updateUserPartial(user);
+
+        //track event and user profile if profile of partialUser changed
+        const newProfile = { name: partialUser.fullName, ...partialUser.additionalData?.profile };
+        if (newProfile) {
+            if (
+                !oldProfile ||
```

Comment:
Here is a commit if you want to :cherries:-pick: https://github.com/gitpod-io/gitpod/commit/b2303343e4dc88f1d3cb99b75750e8b4d3bbb5a1

Reply:
> [this whole section](https://github.com/gitpod-io/gitpod/blob/8bf152ba0304e259d86f28a903192f086dcbec41/components/dashboard/src/settings/ProfileInformation.tsx#L18-L58) should be going into protocol.ts, so it can be used from dashboard and server alike.

makes sense!👍🏼

> Here is a commit if you want to 🍒-pick: https://github.com/gitpod-io/gitpod/commit/b2303343e4dc88f1d3cb99b75750e8b4d3bbb5a1

thanks for the 🍒!🙏🏼 i used `hasChanges` to check before sending events, looks much cleaner now!